### PR TITLE
CP-22536: Translated *.tex files to markdown.

### PR DESCRIPTION
**Pull request by kc284**
These file consisted the API Reference intoduction and coversheet. Removed the
*.tex files that are not used any more and moved the content of the needed ones
to a single markdown file under the doc folder. Replaced the needed *.eps files
with *.png for better handling by markdown. Note that at the moment the cover
sheet still lacks styling and the date as well as the API version are variables.

Signed-off-by: Konstantina Chremmou <konstantina.chremmou@citrix.com>

### CODE REVIEW

**Comment by kc284** [comment]
This first ticket covers the API reference's introduction and coversheet (the latter without any styling at all).
To see what each of the new markdown files produces, one can install pandoc and run
```
pandoc api_ref_intro.md --latex-engine=xelatex --toc --number-sections --variable urlcolor=blue -o intro.pdf
pandoc api_ref_cover.md --latex-engine=xelatex -o cover.pdf
```
One particular thing I would like the reviewer to pay attention to is ensuring I haven't deleted any files needed (some more file deletions will follow in a new PR).

**Comment by lindig** [comment]
I'm not sure these are worth changing: 

* There are code sections where the language is separated from the three backticks by a space.
* I see constructs like `"this"` where `this` probably would have been good enough. 

Both changes can be easily changed globally with an editor.

File: ocaml/doc/pdf/api_ref_intro.md [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+Each field, _f_, has an RPC accessor associated with it
+that returns _f_'s value:
+
+* `"get_f(Ref x)"`: takes a _Ref_ that refers to an object and returns the value
+  of _f_.
+
+Each field, _f_, with attribute _RW_ and whose outermost type is _Set_ has the
+following additional RPCs associated with it:
+
+* `"add_to_f(Ref x, v)"`: adds a new element _v_ to the set referred to by
```
Comment (lindig) [inline ocaml/doc/pdf/api_ref_intro.md]: Here we have `"this"` and I think `this` is good enough.

File: ocaml/doc/pdf/api_ref_intro.md [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+
+API calls are sent over a network to a Xen-enabled host using
+the XML-RPC protocol. In this section we describe how the
+higher-level types used in our API Reference are mapped to
+primitive XML-RPC types.
+
+In our API Reference we specify the signatures of API functions in the following
+style:
+
+``` python
```
Comment (lindig) [inline ocaml/doc/pdf/api_ref_intro.md]: I think the space before `python` must be removed.

File: ocaml/doc/pdf/api_ref_intro.md (hunk 1 above)
Comment (kc284) [inline ocaml/doc/pdf/api_ref_intro.md]: I retained this from the original pdf; I think that was originally added in order to distinguish the real call names (e.g. add_to_guest_agent_config) from pattern call names (add_to_f). I agree it's a bit ugly, so I can remove the quotes if it's obvious what it is.

**Comment by gaborigloi** [comment]
Maybe this is unrelated, but could we also completely remove the `ocaml/idl/xenenterprise_doc_snapshots/` directory?

**Comment by kc284** [comment]
@gaborigloi good idea 👍

**Comment by kc284** [comment]
@gaborigloi I created a differente PR for the old pdfs (PR#3085).

**Comment by robhoes** [comment]
Some, perhaps all, of the intro text is also present in https://xapi-project.github.io/xen-api/basics.html and https://xapi-project.github.io/xen-api/wire-protocol.html, and in markdown format in the xapi-project.github.io repo. It also had some updates and fixes in the text and notations to match the type names etc. of the HTML API docs.

**Comment by kc284** [comment]
So, is it a case of moving it to this repos => autogenerating => uploading to xapi-project.github.io alongisde the json files?

**Comment by kc284** [comment]
I ended up splitting the markdown files again, so we can keep them in this repo and upload them to the xapi project docs. I haven't copied the whole files from the xapi docs, only applied the necessary corrections as I had made some other corrections too. Regarding the vm_lifecyle image, I've decided to keep only the .dot version; other formats can be autogenerated from this one (e.g. png for use on the github docs, eps for pandoc).

**Comment by robhoes** [comment]
I see that you've put the markdown files into ocaml/doc/pdf. Since they can indeed be used to generate a PDF, but it's not restricted to that (HTML is another possibility), perhaps just put them in ocaml/doc instead?

File: ocaml/doc/pdf/basics.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+Each field, `f`, with qualifier `RW` and whose outermost type is `set` has the
+following additional RPCs associated with it:
+
+* `add_f(r, v)`: adds a new element `v` to the set.
+  Note that sets cannot contain duplicate values, hence this operation has
+  no action in the case that `v` is already in the set.
+
+* `remove_f(r, v)`: removes element `v` from the set.
+
+Each field, `f`, with qualifier `RW` and whose outermost type is `set` has the
```
Comment (robhoes) [inline ocaml/doc/pdf/basics.md]: `map`, not `set`.

File: ocaml/doc/pdf/basics.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+following additional RPCs associated with it:
+
+* `add_to_f(r, k, v)`: adds new pair `k -> v`
+  to the mapping stored in `f` in object`r`. Adding a new pair for duplicate
+  key `k`, overwrites any previous mapping for `k`.
```
Comment (robhoes) [inline ocaml/doc/pdf/basics.md]: This is wrong (in the current PDF doc), and was corrected on the github version to read:
```
Adds new pair `(k → v)` to the mapping stored in `f` in object `r`. Attempting to add a new pair for duplicate
key, `k`, fails with an `MAP_DUPLICATE_KEY` error.
```

File: ocaml/doc/pdf/basics.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+* `remove_from_f(r, k)`: removes the pair with key `k`
+  from the mapping stored in `f` in object `r`.
+
+Each field whose outermost type is neither `set` nor `map`, but whose
+qualifier is `RW` has an RPC acessor associated with it that sets its value:
```
Comment (robhoes) [inline ocaml/doc/pdf/basics.md]: accessor

File: ocaml/doc/pdf/basics.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+  with the specified field values.
+
+* Each class has a `get_by_uuid(uuid)` RPC that returns the object
+  of that class that has the specified `uuid`.
+
+* Each class that has a `name_label` field has a
+  `get_by_name_label(name_label)` RPC that returns a set of objects of that
+  class that have the specified `name_label`.
+
+* Each class has a `destroy(r)` RPC that explicitly deletes
```
Comment (robhoes) [inline ocaml/doc/pdf/basics.md]: Most classes

File: ocaml/doc/pdf/wire-protocol.md [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+    </struct>
+```
+
+## Making XML-RPC Calls
+
+### Transport Layer
+
+The following transport layers are currently supported:
+
+* HTTPS for remote administration
```
Comment (robhoes) [inline ocaml/doc/pdf/wire-protocol.md]: Didn't we say that we would make this `HTTP/HTTPS`?

Comment (kc284) [inline ocaml/doc/pdf/wire-protocol.md]: Yes, it seems I reverted it accidentally :(

File: ocaml/doc/pdf/wire-protocol.md [hunk 8]
```diff
@@ -0,0 +1,316 @@
+# Wire Protocol for Remote API Calls
+
+API calls are sent over a network to a Xen-enabled host using
+the [XML-RPC](http://xmlrpc.scripting.com/spec.html) protocol. Here we describe
+how the higher-level types used in our API Reference are mapped to primitive
+XML-RPC types.
+
+We specify the signatures of API functions in the following style:
+
+```python
```
Comment (robhoes) [inline ocaml/doc/pdf/wire-protocol.md]: This (and similar below) isn't Python. Are you just using it to make things look nice in the pandoc output?

Comment (kc284) [inline ocaml/doc/pdf/wire-protocol.md]: Yes, I wasn't sure what other style to use, python seems to be the closest.

**Pull request merged**